src/nvim/screen.c

```diff
@@ -5989,7 +5993,14 @@ void grid_assign_handle(ScreenGrid *grid)
 /// needed.
 void screenalloc(void)
 {
-  static bool entered = false;  // avoid recursiveness
+  // It's possible that we produce an out-of-memory message below, which
+  // will cause this function to be called again.  To break the loop, just
+  // return here.
+  if (resizing) {
+    return;
+  }
+  resizing = true;
+
```

Comment:
would clearing `default_grid.chars`  work instead?

Reply:
Instead of what? `grid_alloc()` will clear the grid as a consequence of reallocating it.

Reply:
instead of the extra `resize` variable. `default_grid.chars == NULL` is checked by `update_screen` already.

Reply:
This would imply moving `VimResized` before `grid_alloc()`, and has other consequences. It could subtly change the behaviour other things (like if VimResized caused a message, will message.c think `printf()` should be used instead of the screen, as there is no screen)?

Also a resize shouldn't by necessity imply that the screen ever has to be deallocated. It would inhibit stuff like possibly use `xrealloc` as an optimization, or a later protocol revision avoiding `grid_clear` on resize (possibly by there being no `default_grid` to start with).